magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,361 @@
+        Array with heights of each cloud layer a.g.l.
+    zd : numpy.float64
+        Zenith distance in deg
+
+    Returns
+    -------
+    numpy ndarray
+        Angular distance in units of degree
+    """
+    d = h / np.cos(zd)
+    return np.arctan((imp / d).to("")).to_value("deg")
+
+
+def model2(imp, h, zd):
+    """
+    Calculates the phenomenological correction to the distances obtained with model0
+
+    Parameters
+    ----------
+    imp : astropy.units.quantity.Quantity
+        Impact
+    h : astropy.units.quantity.Quantity
+        Array with heights of each cloud layer a.g.l.
+    zd : numpy.float64
+        Zenith distance in deg
+
+    Returns
+    -------
+    numpy ndarray
+        Angular distance corrected for bias in units of degrees
+    """
+    H0 = 2.2e3 * u.m
+    bias = 0.877 + 0.015 * ((h + H0) / (7.0e3 * u.m))
+    return bias * model0(imp, h, zd)
+
+
+def trans_height(x, Hc, dHc, trans):
+    """
+    Calculates transmission from a geometrically broad cloud at a set of heights.
+    Cloud is assumed to be homegeneous.
+
+    Parameters
+    ----------
+    x : astropy.units.quantity.Quantity
+        Array with heights of each cloud layer a.g.l.
+    Hc : astropy.units.quantity.Quantity
+        Height of the base of the cloud a.g.l.
+    dHc : astropy.units.quantity.Quantity
+        Cloud thickness in m
+    trans : numpy.float64
+        Transmission of the cloud
+
+    Returns
+    -------
+    numpy.ndarray
+        Cloud transmission
+    """
+    t = pow(trans, ((x - Hc) / dHc).to_value(""))
+    t = np.where(x < Hc, 1, t)
+    t = np.where(x > Hc + dHc, trans, t)
+    return t
+
+
+def process_telescope_data(input_file, config, tel_id, camgeom, focal_eff):
+    """
+    Corrects LST-1 and MAGIC data affected by a cloud presence
+
+    Parameters
+    ----------
+    input_file : str
+        Path to an input .h5 DL1 file
+    config : dict
+        Configuration for the LST-1 + MAGIC analysis
+    tel_id : numpy.int16
+        LST-1 and MAGIC telescope ids
+    camgeom : ctapipe.instrument.camera.geometry.CameraGeometry
+        An instance of the CameraGeometry class containing information about the
+        camera's configuration, including pixel type, number of pixels, rotation
+        angles, and the reference frame.
+    focal_eff : float
+        Effective focal length
+
+    Returns
+    -------
+    pandas.core.frame.DataFrame
+        Data frame of corrected DL1 parameters
+    """
+
+    correction_params = config.get("cloud_correction", {})
+    Hc = u.Quantity(correction_params.get("base_height"))
+    dHc = u.Quantity(correction_params.get("thickness"))
+    trans0 = correction_params.get("vertical_transmission")
+    nlayers = correction_params.get("number_of_layers")
+
+    all_params_list = []
+
+    dl1_params = read_table(input_file, "/events/parameters")
+    dl1_images = read_table(input_file, "/events/dl1/image_" + str(tel_id))
+
+    focal = focal_eff[tel_id]
```

Comment:
you write above that it is a float, but here you treat it as a dictionary (that you indeed pass to the function in main)